### PR TITLE
SC-6612 - i18n Translation theme directory in client

### PR DESCRIPTION
**Pull request by abralurrahman**
# Description

## Links to Tickets or other pull requests

Base links to copy
- https://ticketsystem.hpi-schul-cloud.org/browse/SC-6612

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: theme/thr/views/about/about.hbs [hunk 1]
```diff
@@ -4,12 +4,9 @@
...
-	<p>Die Thüringer Schulcloud soll die technische Grundlage schaffen, dass Lehrkräfte und Schüler in jedem
-		Unterrichtsfach auch moderne digitale Lehr- und Lerninhalte nutzen können, und zwar so, wie Apps über Smartphones
-		oder Tablets nutzbar sind.</p>
-	<p>Die Thüringer Schulcloud ist im Oktober 2019 in die Pilotphase gestartet, in der zunächst 20 Pilotschulen und 9
-		Studienseminare die Thüringer Schulcloud nutzen konnten.</p>
+	<h2 class="h3 section-title center">{{$t "theme.thr.about.headline.aboutTheThuringian"}}</h2>
+	<p>{{$t "theme.thr.about.text.theThuringianSchoolCloudIsIntended"}}</p>
+	<p>{{$t "theme.thr.about.text.theThuringianSchoolCloudStarted"}}</p>
 	<br />
 	<h2 class="h3 section-title center">MINT-Pilotschulen</h2>
```
Comment (ABialoskorski) [inline theme/thr/views/about/about.hbs:11]: please also remember about this

File: theme/thr/views/about/about.hbs [hunk 2]
```diff
@@ -158,25 +155,12 @@
...
-			mit der HPI Schul-Cloud ist eine externe bildungswissenschaftliche Begleitung beauftragt. Zum einen führt
-			die
-			<a href="https://osf.io/bzu4t/?view_only=14ad7519b5a24af7a83cf40b43377142" target="_blank">Universität
-				Tübingen</a> eine wissenschaftliche
-			Befragung der Pilotschulen zu Potenzialen der HPI Schul-Cloud durch. Zum anderen gestaltet das
+		<h2 class="h3 section-title center">{{$t "about.headline.educSupp"}}</h2>
+		<p>{{$t "about.text.forPedagogicalAndDidactic"}}
+			<a href="https://osf.io/bzu4t/?view_only=14ad7519b5a24af7a83cf40b43377142" target="_blank">{{$t "about.text.universityOfTubingen"}}</a> {{$t "about.text.aScientificOne"}}
 			<a href="http://www.gei.de/abteilungen/schulbuch-als-medium/schools-in-the-cloud-wissenschaftliche-begleitforschung-der-schul-cloud-einfuehrung.html"
 				target="_blank">Georg-Eckert-Institut</a>
```
Comment (ABialoskorski) [inline theme/thr/views/about/about.hbs:162]: please also remember about this, "Institut" is diffrent in eng

File: theme/open/views/imprint/imprint.hbs [hunk 3]
```diff
@@ -16,40 +16,19 @@
 		</p>
 	</address>
 	<h2 class="h5">
-		<strong>Vertretungsberechtigter Geschäftsführer:</strong>
+		<strong>{{$t "imprint.imprint_rights.headline.managingDirectorAuthorized" }}</strong>
 	</h2>
 	<p>Prof. Dr. Christoph Meinel
 		<br /> Registergericht: Amtsgericht Potsdam, Registernummer: HRB 12184
 		<br /> Inhaltlich Verantwortlich gemäß § 6 MDStV: Prof. Dr. Christoph Meinel</p>
```
Comment (ABialoskorski) [inline theme/open/views/imprint/imprint.hbs:23]: please remember also about this

File: theme/open/views/imprint/imprint.hbs [hunk 4]
```diff
@@ -16,40 +16,19 @@
...
 	<h2 class="h5">
-		<strong>Redaktion:</strong>
+		<strong>{{$t "imprint.imprint_rights.headline.editorialStuff" }}</strong>
 	</h2>
 	<p>Hasso Plattner Institut</p>
```
Comment (ABialoskorski) [inline theme/open/views/imprint/imprint.hbs:27]: please translate also this "Institut" is diffrent in eng

File: theme/open/views/registration/forms/einwilligungserklaerung.hbs [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
 	<label>
 		<input name="{{prefix}}privacyConsent" type="checkbox" value="true" required/>
-		<span>Ich erkläre mich damit einverstanden, dass die personenbezogenen Daten meines Kindes entsprechend der
+		<span>{{$t "registration.text.iAgreeThatThepersonalData"}}
 			{{#if @root.schoolPrivacyLink}}
-				die <a href="{{@root.schoolPrivacyLink}}" class="base64File-download-btn" id="privacy-link" target="_blank">Datenschutzerklärung</a>
+				die <a href="{{@root.schoolPrivacyLink}}" class="base64File-download-btn" id="privacy-link" target="_blank">{{$t "global.text.dataProtection"}}</a>
 			{{else}}
-				die <a href="{{@root.theme.documents.specificFiles.privacyExemplary}}" target="_blank">Datenschutzerklärung</a>
+				die <a href="{{@root.theme.documents.specificFiles.privacyExemplary}}" target="_blank">{{$t "global.text.dataProtection"}}</a>
```
Comment (ABialoskorski) [inline theme/open/views/registration/forms/einwilligungserklaerung.hbs:14]: die

File: views/about/about.hbs [hunk 6]
```diff
@@ -217,23 +217,10 @@
...
-			mit verfügbarem digitalen Lernmaterial aus der HPI Schul-Cloud und dem Erkenntnisgewinn über die digitale
-			Unterrichtsgestaltung
-			mit der HPI Schul-Cloud ist eine externe bildungswissenschaftliche Begleitung beauftragt. Zum einen führt
-			die
-			<a href="https://osf.io/bzu4t/?view_only=14ad7519b5a24af7a83cf40b43377142" target="_blank">Universität
-				Tübingen</a> eine wissenschaftliche
-			Befragung der Pilotschulen zu Potenzialen der HPI Schul-Cloud durch. Zum anderen gestaltet das
+		<p>{{$t "about.text.forPedagogicalAndDidactic"}}
+			<a href="https://osf.io/bzu4t/?view_only=14ad7519b5a24af7a83cf40b43377142" target="_blank">{{$t "about.text.universityOfTubingen"}}</a> {{$t "about.text.aScientificOne"}}
 			<a href="http://www.gei.de/home.html" target="_blank">Georg-Eckert-Institut</a>
```
Comment (ABialoskorski) [inline views/about/about.hbs:222]: please translate also this "Institut" is diffrent in eng

File: theme/brb/views/authentication/home.hbs [hunk 7]
```diff
@@ -36,49 +36,48 @@
...
+						<li>{{$t "theme.brb.authentication.text.theImportantFunction"}}</li>
+						<li>{{$t "theme.brb.authentication.text.dataProtectionConformity"}}</li>
+						<li>{{$t "theme.brb.authentication.text.theOpportunityToCreate"}}</li>
 					</ul>
 					<br>
 					<p>
-						<img src="https://schul-cloud-hpi.s3.hidrive.strato.com/brb/images/HPI_Schul_Cloud_Brandenburg_Schueler_2019.jpg" alt="Schüler sitzen zusammen und nutzen die Schul-Cloud Brandenburg auf ihren mobilen Endgeräten." />
-						Mehr Informationen: <a href="{{theme.documents.globalFiles.BeschreibungDerSchulCloud}}" target="_blank" rel="noopener">Kurzbeschreibung "Die HPI Schul-Cloud" (.pdf)</a>
+						<img src="https://schul-cloud-hpi.s3.hidrive.strato.com/brb/images/HPI_Schul_Cloud_Brandenburg_Schueler_2019.jpg" alt="{{$t "theme.brb.authentication.img_alt.schoolChildrenSitTogether"}}" />
+						{{$t "theme.brb.authentication.text.moreInformationHPISchoolCloud"}}</a>
```
Comment (ABialoskorski) [inline theme/brb/views/authentication/home.hbs:58]: image

Comment (ABialoskorski) [reply theme/brb/views/authentication/home.hbs:58]: You should use here 3 {

File: theme/brb/views/help/mail-card.hbs [hunk 8]
```diff
@@ -1,2 +1 @@
-<p>Manche Anliegen brauchen mehr Möglichkeiten, als die Felder der Kontaktfunktion zulassen. Schreib uns eine E-Mail, wenn du mit uns ins Gespräch kommen willst. <br/>
-Die richtigen Kontaktpersonen für alle Situationen findest du im <a href="/help/faq/people">Ansprechpartner-Bereich</a>. </p>
+<p>{{$t "theme.brb.help.text.someConcernsRequireMoreOptions"}} </p>
```
Comment (ABialoskorski) [inline theme/brb/views/help/mail-card.hbs:1]: You use <br />, then use {{{ }}}

File: theme/n21/views/imprint/imprint_rights.hbs [hunk 9]
```diff
@@ -13,14 +13,13 @@
 	</p>
 </address>
 <h2 class="h3">
-	Vertretungsberechtigter Geschäftsführer:
+	{{$t "imprint.imprint_rights.headline.managingDirectorAuthorized" }}
 </h2>
 <p>
-	OStD Michael Sternberg
-	<br />Inhaltlich Verantwortlich gemäß § 6 MDStV: OStD Michael Sternberg</p>
+	{{$t "theme.n21.imprint.text.ostd"}}</p>
```
Comment (ABialoskorski) [inline theme/n21/views/imprint/imprint_rights.hbs:19]: use {{{ }}}

File: theme/open/views/authentication/home.hbs [hunk 10]
```diff
@@ -34,8 +34,8 @@
 
 					<section class="card">
 						<div class="card-block">
-							<h2 class="h4">Herzlich willkommen bei der {{theme.title}} !</h2>
-							<p>Das <a href="https://hpi.de/">Hasso-Plattner-Institut</a> für Digital Engineering entwickelt unter der Leitung von Prof. Dr. Christoph Meinel zusammen mit <a href="https://www.mint-ec.de/">MINT-EC</a>, dem nationalen Excellence-Schulnetzwerk von über 300 Schulen bundesweit und unterstützt vom Bundesministerium für Bildung und Forschung die HPI Schul-Cloud. Sie soll die technische Grundlage schaffen, dass Lehrkräfte und Schüler in jedem Unterrichtsfach auch moderne digitale Lehr- und Lerninhalte nutzen können, und zwar so, wie Apps über Smartphones oder Tablets nutzbar sind. Dies ist die offene Instanz der HPI Schul-Cloud, über die einzelne Schulen Zugang zur Plattform erhalten können.</p>
+							<h2 class="h4">{{$t "theme.open.authentication.headline.welcomeToTheTheme" (dict "title" theme.title)}}</h2>
+							<p>{{$t "theme.open.authentication.text.theHassoPlattnerInstituteForDigital"}}</p>
```
Comment (ABialoskorski) [inline theme/open/views/authentication/home.hbs:38]: use {{{ }}}

File: theme/open/views/help/people.hbs [hunk 11]
```diff
@@ -14,12 +14,12 @@
...
 						<div class="card-header" role="tab" id="heading-0">
 							<h2 class="h5 mb-0">
-								Feedback / Wünsche / Probleme bei der Nutzung
+								{{$t "theme.brb.help.headline.feedback/requests"}}
 							</h2>
 						</div>
 						<div id="collapse-0" class="collapse show in">
 							<div class="card-block">
-								Für die Meldung von Problemen oder Wünschen hinsichtlich der Bedienung der HPI Schul-Cloud verwende bitte das Kontaktformular im Hilfebereich oder sende eine E-Mail an <br><a href="mailto:feedback@hpi-schul-cloud.de" target="blank" rel="noopener" class="externalLink">feedback@hpi-schul-cloud.de</a>
+								{{$t "theme.open.help.text.toReportProblemsOrRequestRegarding"}}</a>
```
Comment (ABialoskorski) [inline theme/open/views/help/people.hbs:22]: use {{{ }}}

File: theme/open/views/imprint/imprint.hbs [hunk 12]
```diff
@@ -16,40 +16,19 @@
...
-		publizierten Informationen übernehmen. Das Gleiche gilt für alle Websites, die von den HPI-Seiten über Hyperlinks
-		erreichbar sind. Für alle diese Links gilt: Wir erklären ausdrücklich, dass wir keinen Einfluss auf die Gestaltung
-		und die Inhalte der verlinkten Seite haben. Deshalb distanzieren wir uns hiermit von allen Inhalten aller verlinkten
-		Seiten und machen uns ihre Inhalte nicht zu Eigen. Diese Erklärung gilt für alle auf unserer Seite angebrachten
-		Links zu fremden Seiten. Für den Inhalt dieser verlinkten Seiten ist das HPI nicht verantwortlich. Weiterhin behält
-		sich das HPI jederzeit Änderungen und Ergänzungen der angezeigten Informationen vor. Inhalt und Struktur der
-		HPI-Website sind durch das Copyright geschützt. Die Weiterverwendung von Informationen und Daten ist nur nach
-		unserer Zustimmung zulässig. Wir wünschen Ihnen einen informativen Aufenthalt auf unseren Websites und freuen uns
-		über Vorschläge, Wünsche oder Kritik zur HPI Schul-Cloud.</p>
+		{{$t "theme.open.imprint.text.imprintLegalNoticeOpen"}}</p>
```
Comment (ABialoskorski) [inline theme/open/views/imprint/imprint.hbs:32]: use {{{ }}} and `<p>
{{$t "theme.open.imprint.text.imprintLegalNoticeOpen"}}</p>` should be in the same line

File: theme/open/views/registration/forms/einwilligungserklaerung.hbs [hunk 13]
```diff
@@ -1,30 +1,30 @@
...
+	<p class="explanation">{{{$t "registration._registration.global.text.agreementIsVoluntary" (dict "title" @root.theme.title)}}} {{{$t "registration._registration.global.text.agreementCanBeRevoked"}}} {{{$t "registration._registration.global.text.inCaseOfRevocationDataDeletion"}}}</p>
 {{else}}
 	<label>
 		<input name="{{prefix}}privacyConsent" type="checkbox" value="true" required/>
-		<span>Ich erkläre mich damit einverstanden, dass meine personenbezogenen Daten entsprechend der <a href="{{@root.theme.documents.specificFiles.privacy}}" target="_blank" rel="noopener">Datenschutzerklärung</a> der Schule in der {{@root.theme.title}} sowie bei den von der Schule ausgewählten und weisungsgebundenen Lernmittelanbietern verarbeitet werden.</span>
+		<span>{{{$t "registration._registration.global.text.myDataMayBeProcessed" (dict "link" @root.theme.documents.specificFiles.privacyExemplary "title" @root.theme.title)}}}</span>
```
Comment (ABialoskorski) [inline theme/open/views/registration/forms/einwilligungserklaerung.hbs:22]: should be @root.theme.documents.specificFiles.privacy.
why have You changed to @root.theme.documents.specificFiles.privacyExemplary?

File: theme/open/views/registration/forms/einwilligungserklaerung.hbs [hunk 14]
```diff
@@ -1,30 +1,30 @@
...
+	<p class="explanation">{{{$t "registration._registration.global.text.agreementIsVoluntary" (dict "title" @root.theme.title)}}} {{{$t "registration._registration.global.text.agreementCanBeRevoked"}}} {{{$t "registration._registration.global.text.inCaseOfRevocationDataDeletion"}}}</p>
 {{/if}}
 
 <label>
     <input name="{{prefix}}termsOfUseConsent" type="checkbox" aria-required="true" value="true" required/>
-    <span>Ich habe die <a href="{{@root.theme.documents.specificFiles.termsOfUse}}" target="_blank" rel="noopener">Nutzungsordnung der {{@root.theme.title}}</a> gelesen und stimme ihr zu.</span>
+    <span>{{{$t "registration.text.agreeTermsOfUse" (dict "link" @root.theme.documents.specificFiles.termsOfUseExemplary "title" @root.theme.title)}}}</span>
```
Comment (ABialoskorski) [inline theme/open/views/registration/forms/einwilligungserklaerung.hbs:29]: why @root.theme.documents.specificFiles.termsOfUse changed to @root.theme.documents.specificFiles.termsOfUseExemplary?

File: theme/thr/views/about/about.hbs [hunk 15]
```diff
@@ -158,25 +155,12 @@
...
-				target="_blank">Georg-Eckert-Institut</a>
-			aufbauend auf den Ergebnissen von Unterrichtsbeobachtungen und Workshops mit den Pilotschulen ein
-			Nutzungskonzept für die pädagogisch-didaktisch sinnvolle Anwendung der HPI Schul-Cloud. Außerdem hat das HPI
-			zusammen mit der <a
-				href="https://www.philso.uni-augsburg.de/lehrstuehle/schulpaed/Projekte/Digitalisierung/Lernen-4_0-in-der-Praxis/"
-				target="_blank">Universität Augsburg</a>
-			einen MOOC zu dem Thema <a href="https://lernen.cloud/" target="_blank">Lernen 4.0 -
-				Möglichkeiten und Grenzen einer Digitalisierung im Bildungsbereich</a> durchgeführt.</p>
+				target="_blank">{{$t "theme.thr.about.text.georgEckertInstitute"}}</a>
+			{{$t "about.text.BasedOnTheResults"}}</p>
```
Comment (ABialoskorski) [inline theme/thr/views/about/about.hbs:163]: use {{{ }}}

File: theme/thr/views/help/mail-card.hbs [hunk 16]
```diff
@@ -1,2 +1 @@
-<p>Manche Anliegen brauchen mehr Möglichkeiten, als die Felder des Kontaktformulars zulassen. Schreib uns eine E-Mail, wenn du mit uns ins Gespräch kommen willst.<br />
-Die richtigen Kontaktpersonen für alle Situationen findest du im <a href="/help/faq/people">Ansprechpartner-Bereich</a>.</p>
+<p>{{$t "theme.brb.help.text.someConcernsRequireMoreOptions"}}</p>
```
Comment (ABialoskorski) [inline theme/thr/views/help/mail-card.hbs:1]: use {{{ }}}

File: theme/thr/views/lib/no-create-info.hbs [hunk 17]
```diff
@@ -1,6 +1,3 @@
 <div class="col-sm-12 alert alert-info">
-	<strong>Warum kann ich keine Einträge bearbeiten?</strong><br>
-	Deine Schule in der {{theme.short_title}} ist mit dem Thüringer Schulportal synchronisiert. 
-	Die Daten von Lehrkräften, Schülern und Klassen werden dort zentral verwaltet und können nur dort geändert werden. 
-	Das schließt auch Passwortänderungen mit ein.
+	{{$t "theme.thr.lib.text.whyCantEditEntries"}}
```
Comment (ABialoskorski) [inline theme/thr/views/lib/no-create-info.hbs:2]: use {{{ }}}

File: views/about/about.hbs [hunk 18]
```diff
@@ -217,23 +217,10 @@
...
-			Nutzungskonzept für die pädagogisch-didaktisch sinnvolle Anwendung der HPI Schul-Cloud. Außerdem hat das HPI
-			zusammen mit der <a
-				href="https://www.philso.uni-augsburg.de/lehrstuehle/schulpaed/Projekte/Digitalisierung/Lernen-4_0-in-der-Praxis/"
-				target="_blank">Universität Augsburg</a>
-			einen MOOC zu dem Thema <a href="https://lernen.cloud/" target="_blank">Lernen 4.0 -
-				Möglichkeiten und Grenzen einer Digitalisierung im Bildungsbereich</a> durchgeführt.</p>
+		<p>{{$t "about.text.forPedagogicalAndDidactic"}}
+			<a href="https://osf.io/bzu4t/?view_only=14ad7519b5a24af7a83cf40b43377142" target="_blank">{{$t "about.text.universityOfTubingen"}}</a> {{$t "about.text.aScientificOne"}}
+			<a href="http://www.gei.de/home.html" target="_blank">{{$t "theme.thr.about.text.georgEckertInstitute"}}</a>
+			{{$t "about.text.BasedOnTheResults"}}</p>
```
Comment (ABialoskorski) [inline views/about/about.hbs:223]: use {{{ }}}

**Review by ABialoskorski — CHANGES_REQUESTED** [review]
changes needed

**Comment by ABialoskorski** [comment]
image

File: theme/open/views/registration/forms/einwilligungserklaerung.hbs (hunk 13 above)
Comment (abralurrahman) [reply theme/open/views/registration/forms/einwilligungserklaerung.hbs:22]: fixed :)

File: theme/open/views/registration/forms/einwilligungserklaerung.hbs (hunk 14 above)
Comment (abralurrahman) [reply theme/open/views/registration/forms/einwilligungserklaerung.hbs:29]: fixed ;)

**Review by ABialoskorski — APPROVED** [review]

File: theme/n21/views/help/lern-nuggets.hbs [hunk 19]
```diff
@@ -7,6 +7,6 @@
 	{{/content}}
 
 	{{#content "page"}}
-        Diese Seite ist leider nicht verfügbar.
+        {{$t "theme.brb.help.text.unfortunatelyThisPageIsNotAvailable"}}
```
Comment (janrenz) [inline theme/n21/views/help/lern-nuggets.hbs:10]: is this on purpose?  brb message in n21 theme?

File: theme/open/views/help/people.hbs [hunk 20]
```diff
@@ -14,12 +14,12 @@
 					<div class="card">
 						<div class="card-header" role="tab" id="heading-0">
 							<h2 class="h5 mb-0">
-								Feedback / Wünsche / Probleme bei der Nutzung
+								{{$t "theme.brb.help.headline.feedback/requests"}}
```
Comment (janrenz) [inline theme/open/views/help/people.hbs:17]: brb in open? move this to the general space or copy it.

File: theme/thr/views/help/lern-nuggets.hbs [hunk 21]
```diff
@@ -7,6 +7,6 @@
 	{{/content}}
 
 	{{#content "page"}}
-        Diese Seite ist leider nicht verfügbar.
+        {{$t "theme.brb.help.text.unfortunatelyThisPageIsNotAvailable"}}
```
Comment (janrenz) [inline theme/thr/views/help/lern-nuggets.hbs:10]: use non themed locale here

File: theme/thr/views/help/mail-card.hbs [hunk 22]
```diff
@@ -1,2 +1 @@
-<p>Manche Anliegen brauchen mehr Möglichkeiten, als die Felder des Kontaktformulars zulassen. Schreib uns eine E-Mail, wenn du mit uns ins Gespräch kommen willst.<br />
-Die richtigen Kontaktpersonen für alle Situationen findest du im <a href="/help/faq/people">Ansprechpartner-Bereich</a>.</p>
+<p>{{{$t "theme.brb.help.text.someConcernsRequireMoreOptions"}}}</p>
```
Comment (janrenz) [inline theme/thr/views/help/mail-card.hbs:1]: same here

**Review by janrenz — CHANGES_REQUESTED** [review]

File: theme/n21/views/help/lern-nuggets.hbs (hunk 19 above)
Comment (abralurrahman) [reply theme/n21/views/help/lern-nuggets.hbs:10]: okk. means there should be a general place for those texts which use in several theme?

**Pull request closed without merging**